core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingWithPanacheProcessor.java

```diff
@@ -169,6 +172,52 @@ public void visitMethodInsn(int opcode, String owner, String name, String descri
                     super.visitMethodInsn(Opcodes.INVOKEVIRTUAL, JBOSS_LOGGER_BINARY_NAME, name, descriptor, false);
                 }
 
+                @Override
+                public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootstrapMethodHandle,
+                        Object... bootstrapMethodArguments) {
+
+                    if (!LAMBDA_METAFACTORY.equals(bootstrapMethodHandle.getOwner())) {
+                        super.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, bootstrapMethodArguments);
+                        return;
+                    }
+                    boolean isLogging = false;
+                    for (Object arg : bootstrapMethodArguments) {
+                        if (arg instanceof Handle handle && QUARKUS_LOG_BINARY_NAME.equals(handle.getOwner())) {
+                            isLogging = true;
+                            break;
+                        }
+                    }
+                    if (!isLogging) {
+                        super.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, bootstrapMethodArguments);
+                        return;
+                    }
+
+                    super.visitFieldInsn(Opcodes.GETSTATIC, classNameBinary, SYNTHETIC_LOGGER_FIELD_NAME,
```

Comment:
So this is intercepting *every* method handle of `io.quarkus.logging.Log`, and swapping the invocation from a static to virtual, but keeping it a lambda somehow? Could you explain what's happening in the comments?

Reply:
Yeah, I translate an `invokedynamic` that creates a lambda object of a static call to `Log` to an `invokedynamic` that creates a lambda object of a virtual call to a synthetic `Logger`. Since it's a virtual call, I need the `Logger` on the stack, and need to prepend the `Logger` into the parameter list. I'll add a few comments.